### PR TITLE
Add Google Calendar Action

### PR DESCRIPTION
**Pull request by lspacagna-oai**

### CODE REVIEW

File: examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md [hunk 1]
```diff
@@ -0,0 +1,359 @@
...
+
+- Visit [console.cloud.google.com](https://console.cloud.google.com).
+- In the project selector, choose the project you’d like to use for this GPT Action. If you don’t have a project yet, click the **Create Project** button.
+- When creating a new project, enter a name for it and select the billing account you’d like to associate. In this example, ‘No Organization’ is selected.
+
+<!--ARCADE EMBED START--><div style="position: relative; padding-bottom: calc(55.43981481481482% + 41px); height: 0; width: 100%;"><iframe src="https://demo.arcade.software/0QxHM3NKyUZcv3di9CkA?embed&embed_mobile=tab&embed_desktop=inline&show_copy_link=true" title="Cookbook | Create Google Cloud Project" frameborder="0" loading="lazy" webkitallowfullscreen mozallowfullscreen allowfullscreen allow="clipboard-write" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; color-scheme: light;" ></iframe></div><!--ARCADE EMBED END-->
+
+You now have a Google Cloud Project and are ready to configure the API access to your Google Calendar.
+
+- In the Quck Access menu, select **APIs & Services** > **Library**
```
Comment (pap-openai) [inline examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md:46]: quick?

File: examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md [hunk 2]
```diff
@@ -0,0 +1,359 @@
...
+- Search for **Google Calendar API** (not DKIM) and click on it.
+- Click on the **Enable** button.
+
+<!--ARCADE EMBED START--><div style="position: relative; padding-bottom: calc(53.793774319066145% + 41px); height: 0; width: 100%;"><iframe src="https://demo.arcade.software/uEOZVBdf8OZ8sP0DZAld?embed&embed_mobile=tab&embed_desktop=inline&show_copy_link=true" title="Cookbook | Enable Google Calendar API" frameborder="0" loading="lazy" webkitallowfullscreen mozallowfullscreen allowfullscreen allow="clipboard-write" style="position: absolute; top: 0; left: 0; width: 100%; height: 100%; color-scheme: light;" ></iframe></div><!--ARCADE EMBED END-->
+
+## Setting up OAuth Credentials
```
Comment (pap-openai) [inline examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md:52]: Would "Creating" be better than "setting"? could be confusing if you're setting it up in chatgpt vs setting in up in GCalendar whereas create suggest GCalendar

File: examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md [hunk 3]
```diff
@@ -0,0 +1,359 @@
...
+
+We can now enter the OpenAPI schema for the action. The config below allows reading and creating events. Enter this in the OpenAPI schema field.
+
+```yaml
+openapi: 3.1.0
+info:
+  title: Google Calendar API
+  description: This API allows you to read and create events in a user's Google Calendar.
+  version: 1.0.0
+  contact:
```
Comment (pap-openai) [inline examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md:119]: is "contact" needed?

File: examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md [hunk 4]
```diff
@@ -0,0 +1,359 @@
...
+
+paths:
+  /calendars/primary/events:
+    get:
+      summary: List events from the primary calendar
+      description: Retrieve a list of events from the user's primary Google Calendar.
+      operationId: listEvents
+      tags:
+        - Calendar
+      security:
```
Comment (pap-openai) [inline examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md:135]: I believe security isn't needed as its handled by the GPT

File: images/google-calendar-callback-settings.png
Comment (pap-openai) [inline images/google-calendar-callback-settings.png:1]: would black out client ID, secrets and callback url even if they've been deleted

**Review by pap-openai — APPROVED** [review]
awesome, thanks, left a few minor comments, feel free to merge after you've incorporated some of the feedback

File: examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md (hunk 1 above)
Comment (lspacagna-oai) [reply examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md:46]: Yep thats what its titled in the Google UI!

File: examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md (hunk 2 above)
Comment (lspacagna-oai) [reply examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md:52]: Yep, will change to creating!

File: examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md (hunk 4 above)
Comment (lspacagna-oai) [reply examples/chatgpt/gpt_actions_library/gpt_action_google_calendar.md:135]: Good spot, removed.

File: images/google-calendar-callback-settings.png
Comment (lspacagna-oai) [reply images/google-calendar-callback-settings.png:1]: Nice spot, one that I missed!

**Comment by mansiibm** [comment]
hey thankyou for this awesome gpt action and openai spec in the documentation, I had one query that using this i am able to send the google invites to only google domain mails and not outlook ones, can you please help in here. Using exactly as mentioned in the docs- https://cookbook.openai.com/examples/chatgpt/gpt_actions_library/gpt_action_google_calendar .

My concern is to send via actions to outlook also. Thanks in advance, sorry for commenting here , could not reach out in other way.

**Pull request merged**